compiler-builtins/src/thumbv6k.rs

```diff
@@ -0,0 +1,200 @@
+// Armv6k supports atomic instructions, but they are unavailable in Thumb mode
+// unless Thumb-2 instructions available (v6t2).
+// Using Thumb interworking allows us to use these instructions even from Thumb mode
+// without Thumb-2 instructions, but LLVM does not implement that processing (as of LLVM 21),
+// so we implement it here at this time.
+
+use core::arch::asm;
+use core::mem;
+
+// Data Memory Barrier (DMB) operation.
+//
+// Armv6 does not support DMB instruction, so use use special instruction equivalent to it.
+//
+// Refs: https://developer.arm.com/documentation/ddi0360/f/control-coprocessor-cp15/register-descriptions/c7--cache-operations-register
+macro_rules! dmb {
+    () => {
+        "mcr p15, #0, {zero}, c7, c10, #5"
+    };
+}
+macro_rules! asm_use_dmb {
```

Comment:
I think this macro is just an asm block, but with macro expansion (and an extra `{zero}` register which I guess is useful somewhere). But it could probably do with a comment to explain all that because it's really non-obvious. Especially as this macro doesn't have anything to do with `dmb` (other than it's written as a macro that needs to be expanded).

Reply:
This was originally a helper for [writing code that required no changes other than using the Data Memory Barrier (DMB instruction or equivalent instructions)](https://github.com/taiki-e/atomic-maybe-uninit/blob/f98d8ab43c83e49bc3e030d8c3cb8ea1c5f96cab/src/arch/arm.rs#L187-L203).

In this PR, it is indeed unclear, so it can be deleted.